### PR TITLE
OPSEXP-1046 java base multistage build

### PR DESCRIPTION
**Pull request by alxgomz**
now creates multiple distributions images and also a jdk based one (centos7 only) mainly for backward compatibility purposes.
Moves build process to github action.
changes naming scheme as described in README.md
* STARTS CREATING NEW TAGS
* STOPS CREATING TAGS WITH PREVIOUS NAMING (children images will need to align)

### CODE REVIEW

**Review by Mieszko96 — COMMENTED** [review]
It looks good to me, but let someone else take a look.

**Review by EquinoxBlack — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,72 @@
...
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+RUN mkdir -p /usr/share/man/man1 || true; \
+    # Update here in case of java upgrade
+    # check https://ubuntu.com/security/cves?q=&package=openjdk-12 for security updates
+    JAVA_PKG_VERSION=11.0.13+8-0ubuntu1~20.04; \
+    DEBIAN_FRONTEND=noninteractive; \
+    apt-get update -qq && apt-get upgrade -qq && \
+    apt-get install --no-install-recommends -qq openjdk-11-${JDIST}-headless=${JAVA_PKG_VERSION} && \
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply Dockerfile:20]: My understanding is we want to stick to the supported platform as much as possible, hence the pining policy.
In reality following the security updates already forces us to deviate from the supported stack (e.g. we now ship 11.0.3 whilesupported stackoffers 11.0.7 only)
@mteodori what do you think? Should we keep a explicit upgrade policy or always install latest when generating new image? (in particular for java)

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+ENV JAVA_HOME=/usr/lib/jvm/java
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+RUN mkdir -p /usr/share/man/man1 || true; \
+    # Update here in case of java upgrade
+    # check https://ubuntu.com/security/cves?q=&package=openjdk-12 for security updates
+    JAVA_PKG_VERSION=11.0.13+8-0ubuntu1~20.04; \
+    DEBIAN_FRONTEND=noninteractive; \
+    apt-get update -qq && apt-get upgrade -qq && \
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply Dockerfile:19]: true, but if we don't do that none of our image would give us back a green security scan. We consider basic system libs to be not impactful enough for our java applications to allow auto upgrades. Though I agree this assumption can be proven wrong, but it's either that or deal manually with all packages versions manually (ajva or tomcat are different beasts)

File: .github/dependabot.yml [hunk 3]
```diff
@@ -6,10 +6,14 @@
 version: 2
 updates:
   - package-ecosystem: "docker"
-    directory: "/centos-7/"
+    directory: "/"
     schedule:
       interval: "daily"
     ignore:
+      - dependency-name: "ubuntu"
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply .github/dependabot.yml:13]: no good  reason I guess. fixed

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mteodori) [reply Dockerfile:20]: package latest of java major version, supported platforms is being updated to just include that

**Pull request merged**